### DISCUSSION TITLE
Specificity issues with a third party library

### DISCUSSION THREAD

**Discussion by martinszeltins** [tailwindlabs/tailwindcss#8117] [Help]
So I have this 3rd party component library that I'm using. But the `@tailwind base` is overriding the form styles and has a higher specificity than the 3rd party library. If I put the 3rd party lib CSS later than `tailwind` then I cannot override the styles. Let me show you what I mean...

I have these in my HTML:

````
import '@/assets/css/vendor/3rd-party-lib-theme.css'
import '@/assets/css/app.css'
````

Inside `app.css` I have this:

````
@tailwind base;
@tailwind components;
@tailwind utilities;
````

Then I have this button in my HTML:

````
<button class="p-button p-component">
````

The 3rd party library has some style for this button like `background: purple` but since my `tailwind` import comes **after** my 3rd party lib import then these Tailwind style override the lib styles like this:

````
// Tailwind styles here override the .p-button styles :(
button, [type='button'] {
    background-color: transparent;
    background-image: none;
}

.p-button {
    color: #ffffff;
    background: #8B5CF6;
}
````

The result is that the button gets a transparent background :(

So I thought, ok - no big deal, I will just put my 3rd party css import **after** the Tailwind import. But then my 3rd party lib has higher specificity and I cannot override those styles.

````
import '@/assets/css/app.css'
import '@/assets/css/vendor/3rd-party-lib-theme.css'
````

Now this happens - here my `bg-blue-300` class is ignored because the 3rd party lib has higher specificity since it comes later in CSS.

````
<button class="p-button p-component bg-blue-300" type="button">
  Log in
</button>
````

**Comment by mryechkin** [comment]
Hey @martinszeltins - were you able to solve this issue?

I'm running into the exact same problem as well and wondering if you ended up finding a solution.

**Reply by martinszeltins** [reply]
I think I solved it by giving a higher specificity for my 3rd party library overrides and then it worked. What is your specific case?
